src/frontend/src/optimizer/rule/apply_scan.rs

```diff
@@ -33,7 +33,11 @@ impl Rule for ApplyScanRule {
 
         // LogicalJoin with correlated inputs in join condition has been handled by ApplyJoin. This
         // handles the ones without correlation
-        if let (None, None) = (right.as_logical_scan(), right.as_logical_join()) {
+        if let (None, None, None) = (
+            right.as_logical_scan(),
+            right.as_logical_join(),
+            right.as_logical_values(),
```

Comment:
To be fully correct, the exprs in `values` should not contain correlated input refs. Just raising awareness of this rather than suggesting an immediate fix, as we do not consider correlation in `values` at binding phase either:

https://github.com/singularity-data/risingwave/blob/3b84c1653d99c0a31b3326b7dca71d474cf9decc/src/frontend/src/binder/set_expr.rs#L40-L57

In the future, ApplyValues may look closer to ApplyProj rather than ApplyScan.

Reply:
So we should report error in binder when BoundValues has `correlated_input_refs`?